### ISSUE TITLE
Base master not found; no master at default location?

### ISSUE THREAD

**Issue by JasperTerra** [googlei18n/fontmake#401]
I’m trying to generate a variable font with its weight axis mapped, however I get the following error:
```
Traceback (most recent call last):
  File "/usr/local/bin/fontmake", line 9, in <module>
    load_entry_point('fontmake==1.4.0', 'console_scripts', 'fontmake')()
  File "/Users/jasperterra/Documents/fontmake/Lib/fontmake/__main__.py", line 191, in main
    project.run_from_designspace(designspace_path, **args)
  File "/Users/jasperterra/Documents/fontmake/Lib/fontmake/font_project.py", line 437, in run_from_designspace
    interpolate_layout_from=interpolate_layout_from, **kwargs)
  File "/Users/jasperterra/Documents/fontmake/Lib/fontmake/font_project.py", line 496, in run_from_ufos
    self.build_variable_font(designspace_path)
  File "/Users/jasperterra/Documents/fontmake/Lib/fontmake/font_project.py", line 220, in build_variable_font
    font, _, _ = varLib.build(designspace_path, finder)
  File "/Users/jasperterra/Documents/fontmake/src/fonttools/Lib/fontTools/varLib/__init__.py", line 691, in build
    axes, internal_axis_supports, base_idx, normalized_master_locs, masters, instances = load_designspace(designspace_filename)
  File "/Users/jasperterra/Documents/fontmake/src/fonttools/Lib/fontTools/varLib/__init__.py", line 676, in load_designspace
    assert base_idx is not None, "Base master not found; no master at default location?"
AssertionError: Base master not found; no master at default location?
```
I have a folder with six ufo masters, a designspace file (two axes; weight and width) and a stat.ttx file. When the weight axis is not mapped and I run fontmake all works as expected (working variable font). The default source (Regular) has the following added:
```
           <lib copy="1" />
            <groups copy="1" />
            <info copy="1" />
```
Mapping of the weight axes looks like this:
```
            <map input="100" output="-925" />
            <map input="200" output="-706" />
            <map input="300" output="-360" />
            <map input="400" output="0" />
            <map input="450" output="191" />
            <map input="500" output="427" />
            <map input="600" output="705" />
            <map input="700" output="1000" />
```
Where/how does one set the default location and its base master in the designspace file?

**Comment by anthrotype** [comment]
the base master is the one which is at the default location (0, in normalized coordinates) on all the axes.

can you share your designspace file?

**Comment by JasperTerra** [comment]
Sure! 
[IBMPlexSans-Roman.designspace.zip](https://github.com/googlei18n/fontmake/files/1741174/IBMPlexSans-Roman.designspace.zip)

**Comment by miguelsousa** [comment]
@JasperTerra the definition of the `weight` axis should be this instead:
```xml
<axis default="400" maximum="700" minimum="100" name="weight" tag="wght">
```
The `width` axis will probably need a similar change as well.

**Comment by JasperTerra** [comment]
That did the trick. Thanks guys!

**Comment by moyogo** [comment]
@JasperTerra Side note: The OpenType Specification requires the Regular to have the value 100 in the `wdth`.
See https://docs.microsoft.com/en-us/typography/opentype/spec/dvaraxistag_wdth:
> Recommended or required "Regular" value: 100 is required.

You should also provide a mapping from `input="100" output="0"` for the `wdth` axis.

**Comment by JasperTerra** [comment]
Cool! Thanks.

**Comment by anthrotype** [comment]
Thanks @miguelsousa and @moyogo for debugging this

**Comment by caiosimonbreda** [comment]
Hello, 

I'm having the same problem described on this issue when trying to build a variable TT font using Fontmake. I worked on the masters using Fontlab 7, and don't quite get why the Regular master isn't found/recognised. Perhaps something to do with the custom axes? 


[grima.designspace.zip](https://github.com/googlefonts/fontmake/files/6571812/grima.designspace.zip)

**Comment by tiroj** [comment]
I am hitting the same error, and it is doing my head in. I have checked over the designspace file and UFOs again and again, and cannot spot a problem. This is the designspace file:

```
<?xml version='1.0' encoding='utf-8'?>
<designspace format="4.0">
 <axes>
  <axis default="400" tag="wght" name="weight" minimum="400" maximum="700">
   <labelname xml:lang="en">Weight</labelname>
   <map input="400" output="174"/>
   <map input="500" output="210"/>
   <map input="600" output="252"/>
   <map input="700" output="304"/>
  </axis>
 </axes>
 <sources>
  <source filename="SI-Bold.ufo" familyname="SI" stylename="Bold" name="Bold">
   <location>
    <dimension xvalue="304" name="weight"/>
   </location>
  </source>
  <source filename="SI-Regular.ufo" familyname="SI" stylename="Regular" name="Regular">
   <lib copy="1"/>
   <groups copy="1"/>
   <info copy="1"/>
   <features copy="1"/>
   <location>
    <dimension xvalue="174" name="weight"/>
   </location>
  </source>
 </sources>
 <instances>
  <instance filename="instances/SI-Regular.ufo" familyname="SI" stylename="Regular" name="Regular">
   <location>
    <dimension xvalue="174" name="weight"/>
   </location>
  </instance>
  <instance filename="instances/SI-Medium.ufo" familyname="SI" stylename="Medium" name="Medium">
   <location>
    <dimension xvalue="210" name="weight"/>
   </location>
  </instance>
  <instance filename="instances/SI-Semibold.ufo" familyname="SI" stylename="Semibold" name="Semibold">
   <location>
    <dimension xvalue="252" name="weight"/>
   </location>
  </instance>
  <instance filename="instances/SI-Bold.ufo" familyname="SI" stylename="Bold" name="Bold">
   <location>
    <dimension xvalue="304" name="weight"/>
   </location>
  </instance>
 </instances>
</designspace>
```
The UFO master file names all correspond to what is in the designspace file, but I am getting this error both when I run my build tool and also if I try to open the designspace file in FontGoggles.

`fontTools.varLib.errors.VariationModelError: Base master not found.`

Anyone spot the problem or have a guess?

**Comment by tiroj** [comment]
Apologies for the late follow-up. I tracked this error down to incompatible anchor data between masters (which was in turn due to a bug in FontLab). Fixing the anchor compatibility resolved the build issue.

I have now encountered the same error in another project, and am guessing it is a similar anchor problem, but really the fontTools error message could be more helpful, as this doesn’t seem to be obvious a ‘Base master not found.’ issue.
